doc/designs/quic-design/rx-depacketizer.md

```diff
@@ -0,0 +1,260 @@
+[Table 1](#table-1) below.
+
+#### Table 1
+
+Taken from [RFC 9000 12.4 Frames and Frame Types]
+
+| Type |           Name           |         Passed to         | ACK eliciting |
+| ---- | ------------------------ | ------------------------- | ------------- |
+| 0x00 | [padding]                | -                         |               |
+| 0x01 | [ping]                   | -                         | &#10004;      |
+| 0x02 | [ack 0x02]               | [ACK manager] [^1]        |               |
+| 0x03 | [ack 0x03]               | [ACK manager] [^1]        |               |
+| 0x04 | [reset_stream]           | - [^2]                    | &#10004;      |
+| 0x05 | [stop_sending]           | - [^3]                    | &#10004;      |
+| 0x06 | [crypto]                 | Handshake manager         | &#10004;      |
+| 0x07 | [new_token]              | Session manager           | &#10004;      |
+| 0x08 | [stream 0x08]            | Apprioriate stream [^4]   | &#10004;      |
+| 0x09 | [stream 0x09]            | Apprioriate stream [^4]   | &#10004;      |
+| 0x0A | [stream 0x0A]            | Apprioriate stream [^4]   | &#10004;      |
+| 0x0B | [stream 0x0B]            | Apprioriate stream [^4]   | &#10004;      |
+| 0x0C | [stream 0x0C]            | Apprioriate stream [^4]   | &#10004;      |
+| 0x0D | [stream 0x0D]            | Apprioriate stream [^4]   | &#10004;      |
+| 0x0E | [stream 0x0E]            | Apprioriate stream [^4]   | &#10004;      |
+| 0x0F | [stream 0x0F]            | Apprioriate stream [^4]   | &#10004;      |
+| 0x10 | [max_data]               | Flow control [^5]         | &#10004;      |
+| 0x11 | [max_stream_data]        | Flow control [^5]         | &#10004;      |
+| 0x12 | [max_streams 0x12]       | Connection manager? [^6]  | &#10004;      |
+| 0x13 | [max_streams 0x13]       | Connection manager? [^6]  | &#10004;      |
+| 0x14 | [data_blocked]           | Flow control [^5]         | &#10004;      |
+| 0x15 | [stream_data_blocked]    | Flow control [^5]         | &#10004;      |
+| 0x16 | [streams_blocked 0x16]   | Connection manager? [^6]  | &#10004;      |
+| 0x17 | [streams_blocked 0x17]   | Connection manager? [^6]  | &#10004;      |
+| 0x18 | [new_connection_id]      | Connection manager        | &#10004;      |
+| 0x19 | [retire_connection_id]   | Connection manager        | &#10004;      |
+| 0x1A | [path_challenge]         | Connection manager? [^7]  | &#10004;      |
+| 0x1B | [path_response]          | Connection manager? [^7]  | &#10004;      |
+| 0x1C | [connection_close 0x1C]  | Connection manager        |               |
+| 0x1D | [connection_close 0x1D]  | Connection manager        |               |
+| 0x1E | [handshake_done]         | Handshake manager         | &#10004;      |
+| ???? | *[Extension Frames]*     | - [^8]                    | &#10004;      |
+
+[ the table used here is shamelessly copied from #18570 ]
+
+Notes:
+
+[^1]: This creates and populates an `QUIC_ACKM_ACK` structure, then calls
```

Comment:
End notes are painful to look up.  These might be better inline as Notes sections.

Reply:
?  This is a notes section.

Reply:
Or, not sure what you mean with inlined...  do you mean that you want a notes column, with the same text repeated for every reference?  Eh gods, that's going to turn out ugly...

Reply:
Read these notes, decide which are important at the point they are first referenced.
Those should be included at the point of first reference.

I believe most are relevant.

Footnotes/end notes are for things readers can skip without impacting comprehension of the material being read.

Reply:
I disagree